samcli/lib/intrinsic_resolver/intrinsics_symbol_table.py

```diff
@@ -363,8 +364,14 @@ def handle_pseudo_region(self):
         -------
         The region from the environment or a default one
         """
+        try:
+            ctx = Context.get_current_context()
```

Comment:
Context is specific for Click and the command line, why are we adding it into the lib? 

Can we fix this further up the chain? It appears we set the Region here: https://github.com/aws/aws-sam-cli/blob/199ded1901964ed17023e9d5ed5682ff77ae3dfd/samcli/lib/providers/sam_base_provider.py#L196 and is expected to be overridden here: https://github.com/aws/aws-sam-cli/blob/199ded1901964ed17023e9d5ed5682ff77ae3dfd/samcli/lib/providers/sam_base_provider.py#L166 which would trigger line 373 here.

This is very much expected behavior looking at the comments above: https://github.com/aws/aws-sam-cli/blob/199ded1901964ed17023e9d5ed5682ff77ae3dfd/samcli/lib/intrinsic_resolver/intrinsics_symbol_table.py#L103